internal/pkg/mcp/controller.go

```diff
@@ -0,0 +1,203 @@
+package mcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	api_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const maxRetries = 5
+
+var (
+	serviceController *Controller
+)
+
+// Event indicate the informerEvent
+type Event struct {
+	key          string
+	eventType    string
+	namespace    string
+	resourceType string
+	obj          runtime.Object
+	oldObj       runtime.Object
+}
+
+// Controller object
+type Controller struct {
+	clientset    kubernetes.Interface
+	queue        workqueue.RateLimitingInterface
+	informer     cache.SharedIndexInformer
+	eventHandler Handler
+	resourceType string
+}
+
+func objName(obj interface{}) string {
+	return reflect.TypeOf(obj).Name()
+}
+
+func StartControllers(ctx context.Context) {
```

Comment:
In general, we will not need informers to handle imported service, because they will come from the remote server and won't be created as local `Service` objects.

Reply:
the idea is just to rely on the informer as a cache for local `Service`s, which we query to know whether we need to create a `ServiceEntry` or not.

Reply:
Ah right!